packages/lib/slots.ts

```diff
@@ -160,19 +160,52 @@ function buildSlotsWithDateRanges({
   frequency = minimumOfOne(frequency);
   eventLength = minimumOfOne(eventLength);
   offsetStart = offsetStart ? minimumOfOne(offsetStart) : 0;
-
   const slots: { time: Dayjs; userIds?: number[] }[] = [];
   dateRanges.forEach((range) => {
     const startTimeWithMinNotice = dayjs.utc().add(minimumBookingNotice, "minute");
 
     let slotStartTime = range.start.isAfter(startTimeWithMinNotice) ? range.start : startTimeWithMinNotice;
 
-    slotStartTime =
-      slotStartTime.utc().minute() % 15 !== 0
-        ? slotStartTime
-            .startOf("day")
-            .add(slotStartTime.hour() * 60 + Math.ceil(slotStartTime.minute() / 15) * 15, "minute")
-        : slotStartTime;
+    let previousStartTime;
+    // not sure here (we want to check if it is the same day in the organizer's timezone)
+    if (
+      slots.length &&
+      dayjs
+        .utc(range.start)
+        .add(range.start.utcOffset())
+        .isSame(dayjs.utc(slots[slots.length - 1].time).add(slots[slots.length - 1].time.utcOffset()), "day")
```

Comment:
```suggestion
 range.start.isSame(slots[slots.length - 1].time, "day")
```
@emrysal Just to make sure, this wouldn't work, right? I think `isSame' would compare in the system timezone but we want to check if it is the same day in the timezone of the organizer

Reply:
range start is various timezones, so wouldn't no